src/Workspaces/VisualBasic/Portable/Utilities/SpeculationAnalyzer.vb

```diff
@@ -420,6 +425,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Utilities
             Return False
         End Function
 
+        Private Function ReplacementBreaksWithStatement(originalWithStatement As WithStatementSyntax, replacedWithStatement As WithStatementSyntax) As Boolean
+            Dim originalExpression = originalWithStatement.Expression
+            Dim replacedExpression = replacedWithStatement.Expression
+            Select Case originalExpression.Kind
```

Comment:
You could simplify this. You already have the expression syntax nodes and `SemanticModel.GetTypeInfo` doesn't need it to be cast to a base type in for it to work. We would hope that no matter what type of expression we are dealing with it wouldn't break based upon the change.

Reply:
Good call!  Definitely don't need to cast to the CastExpressionSyntax or PredefinedCastExpressionSyntax.

Additionally according to the docs the [with statement must always evaluate to an object](https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/with-end-with-statement), so it would make sense that the evaluation of the type should result the same value in the original and replacement node.